### ISSUE TITLE
Parsing percentage status (no new lines)

### ISSUE THREAD

**Issue by hulaomr** [Tyrrrz/CliWrap#71]
Hi again Alexey,
Please, may you suggest an approach using CliWrap to parse the status as show the application below:
![status](https://user-images.githubusercontent.com/36808114/84787568-6f251580-afee-11ea-924c-9d7c61e07af3.gif)
Such status will be show in a WinForm.
I tried something similar to the following method  
`public async Task I_can_execute_a_command_that_pipes_its_stdout_into_an_async_delegate()` at [PipingSpec.cs](https://github.com/Tyrrrz/CliWrap/blob/master/CliWrap.Tests/PipingSpecs.cs), but I just get the calling when there is a new line, at the end of the status, i.e. 100%.
Any guidance on which approach use is appreciated.
Thanks

**Comment by Tyrrrz** [comment]
Hi.
I've been able to work around this through a custom pipe target. Here's an example:

https://github.com/Tyrrrz/YoutubeExplode.Converter/blob/de02be0fa39d8655ee76c2ff3ce82795240e41a9/YoutubeExplode.Converter/Internal/FFmpeg.cs#L69-L119

This is not clean and i'm still looking for a better solution.

**Comment by hulaomr** [comment]
Thanks for sending me the reference using a custom pipe. I will need to take a look more deeply in the library.

**Comment by hulaomr** [comment]
Hi Alexey,

After reviewing the custom pipe target, I see the class method **CopyFromAsync** is using a **StreamReader** for the stream passed by the library, with the specified character encoding, without **byte order mark detection**, buffer size of 1024 and **leaving the stream open**.

Based on such proposal, I implemented the following function to run the application:

```

        public async System.Threading.Tasks.Task<int> UpdateProcessAsync()
        {
            var cts = new CancellationTokenSource();
            cts.CancelAfter(TimeSpan.FromSeconds(30));

            var cmd = Cli.Wrap(toolFilePath).WithArguments(new[] { "arg1", "arg2" }).WithStandardOutputPipe(new CustomPipeTarget());

            var result = cmd.ExecuteAsync(cts.Token);

            return 0;
        }
```

And the Custom Pipe, based on FFmpegProgressRouter as below:

```
            private class CustomPipeTarget : PipeTarget
            {
                public CustomPipeTarget()
                {

                }

                public override async Task CopyFromAsync(Stream source, CancellationToken cancellationToken = default)
                {
                    var reader = new StreamReader(source, Console.OutputEncoding, false, 1024, true);

                    var buffer = new char[1024];
                    int charsRead;

                    while ((charsRead = await reader.ReadAsync(buffer, 0, buffer.Length)) > 0)
                    {
                        cancellationToken.ThrowIfCancellationRequested();
                        Console.WriteLine(charsRead);
                    }
                }
            }
```

The method **CopyFromAsync** is called but the results was the same. Only read the characters when the process finish with a message "Process finished" plus the End of Line character.

Please, may you confirm if the application **fmpeg** is not generating a end of line character. 

I starting thinking the third party application I'm using is not flushing the data to the standard output, just showing in the screen with methods similar to [theses](https://stackoverflow.com/questions/888533/how-can-i-update-the-current-line-in-a-c-sharp-windows-console-app). 

In any case, thanks for your support in this matter. I will pass the page and try to look for additional arguments in order to get more verbose output.

**Comment by Tyrrrz** [comment]
Are you sure the progress is not written to stderr instead?

**Comment by hulaomr** [comment]
Yes. 

In any case, I tried with **WithStandardErrorPipe** as well.

**Comment by hulaomr** [comment]
Alexey,
As mentioned previously, I enabled others arguments to obtain more info (verbose) from the third party application and show some kind of status to the user.
I used **Custom Pipe** and **IProgress** as proposed in [FFmpeg.cs](https://github.com/Tyrrrz/YoutubeExplode.Converter/blob/de02be0fa39d8655ee76c2ff3ce82795240e41a9/YoutubeExplode.Converter/Internal/FFmpeg.cs#L69-L119).
At the end, I was able to obtain the output only when there were new lines characters. I got similar results with and without CliWrap.
Thanks!
